nighthawk/include/nighthawk/common/statistic.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include <cstdint>
+#include <memory>
+#include <string>
+
+#include "common/common/non_copyable.h"
+#include "envoy/common/exception.h"
+#include "envoy/common/pure.h"
+
+#include "nighthawk/source/client/output.pb.h"
+
+namespace Nighthawk {
+
+class StatisticException : public Envoy::EnvoyException {
+public:
+  StatisticException() : Envoy::EnvoyException("StatisticException") {}
+};
+
+/**
+ * Abstract interface for a statistic.
+ */
+class Statistic : Envoy::NonCopyable {
+public:
+  virtual ~Statistic() = default;
+  /**
+   * Method for adding a sample value.
+   * @param value the value of the sample to add
+   */
+  virtual void addValue(int64_t sample_value) PURE;
+
+  virtual uint64_t count() const PURE;
+  virtual double mean() const PURE;
+  virtual double pvariance() const PURE;
+  virtual double pstdev() const PURE;
+
+  /**
+   * Only used in tests to match expectations to the right precision level.
+   * @return Number of significant digits. 0 is assumed to be max precision.
+   */
+  virtual uint64_t significantDigits() const { return 0; }
+
+  /**
+   * Indicates if the implementation is subject to catastrophic cancellation.
+   * Used in tests.
+   * @return True iff catastrophic cancellation should not occur.
+   */
+  virtual bool resistsCatastrophicCancellation() const { return false; }
+
+  /**
+   * Gets a representation of the statistic as a std::string.
```

Comment:
Nit: prefer one liner `@return std::string a representation of ..` style for these.